torch/_inductor/codegen/common.py

```diff
@@ -511,6 +511,185 @@ def remainder(a, b):
     def load_seed(name, offset):
         return ops.load(name, sympy.Integer(offset))
 
+    @classmethod
+    def _initialize_pointwise_overrides(cls, target):
+        assert target in {"triton", "cpp", "cppvec"}, target
+
+        def pointwise_factory_1(impl):
+            def func(x):
+                return impl.format(x=x)
+
+            return func
+
+        def pointwise_factory_2(impl):
+            def func(x, y):
+                return impl.format(x=x, y=y)
+
+            return func
+
+        for funcname, impls in pointwise_overrides_data.items():
+            impl = impls.get(target)
+            if impl is not None:
+                nof_args = 2 if "{y}" in impl else 1
+                # extend the following dictionary with factory
+                # functions for a specific number of arguments as
+                # needed:
+                factory = {1: pointwise_factory_1, 2: pointwise_factory_2}[nof_args]
+                setattr(cls, funcname, staticmethod(factory(impl)))
+
+
+pointwise_overrides_data: Dict[str, Dict[str, str]] = dict(
```

Comment:
This is good as-is, but I think we could expand this to non-numeric operators as well if you changed `Dict[str, str]` to a `dataclass` and had `type_promotion_kind` be a member.

Reply:
Are you suggesting introducing a new dataclass that holds `cpp`, `triton`, `cppvec` members instead of using `dict`? If so, how adding another member `type_promotion_kind` will relate to supporting `non-numeric` operators?

Reply:
Instead of calling `register_pointwise_numeric` in `lowering.py`, you could use `register_pointwise` and pass the `type_promotion_kind`. This would allow registering operators that don't promote int to float.

Reply:
Ok, gotcha you.

Reply:
I introduced `type_promotion_kind` item and switched to using `Dict[str, Any]` for now.

Reply:
`dataclass` would be better so we could get strong typing. There are ongoing efforts to make mypy typing more strict throughout inductor.

Reply:
ok, I'll give it a shot.

Reply:
done